File: DTO/Cache/Models/UserModel.cs

namespace DTO.Cache.Models {

	public class UserModel {

		public int UserId { get; set; }
		public string UserName { get; set; }
		public string Password { get; set; }
		public int UserRoleId { get; set; }
		public UserRoleModel UserRole { get; set; }
		public PersonModel Person { get; set; }
		public bool IsActive { get; set; }
	}
}

File: AngularClient/src/app/ts/classes/models/user.model.ts
import { UserI, UserRoleI, PersonI } from '../../interfaces/model.interfaces';
import { UserRoleTypes } from '../../enums/user-role-types.enum';

export class User implements UserI {
	public userId: number
	public userName: string
	public isActive: boolean
	public password: string
	public userRoleId: UserRoleTypes
	
	constructor(
		public userRole: UserRoleI,
		public person: PersonI,
	) { }
	
	public assignAllData(data: Object): UserI {
		try {
			return Object.assign(this, data)
		} catch (e) {
			console.error("Can't assign data at: ", this)
			console.error("Data: ", data)
			console.error(e)
			return null
		}
	}
}